File: backend/rest-server/models/transaction.js
/**
 * Transaction model.
 */
var mongoose = require('mongoose');

//Define schema
var TransactionSchema = mongoose.Schema({
    user: {type: String, required:false},
    business_owner: {type: String, required: true},
    timestamp: {type: Number, required: true},
    virtual_money_used: {type: Number, required: false},
    payback_amount: {type: Number, required: false},
    total_amount: {type: Number, required: false},
    status: {type: String, required: true}
});



//Compile model
Transaction = mongoose.model('Transaction', TransactionSchema);

//Export the model
module.exports = Transaction;

File: lime-java-modules/lime-model/src/main/java/edu/upc/fib/bip/lime/model/Transaction.java
package edu.upc.fib.bip.lime.model;

import lombok.*;

import java.io.Serializable;

/**
 * @author Elizaveta Ketova <elizabeth.ooh@gmail.com>
 * @since 02.12.17
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Transaction implements Serializable {

    private String user;
    private String business_owner;
    private long timestamp;
    private double virtual_money_used;
    private double payback_amount;
    private double total_amount;
    private String status = "confirmed";

}
